### ISSUE TITLE
Fix binomial distribution of parameters in a for loop when using a boolean hyperparameter

### ISSUE THREAD

**Issue by Marko-Shevchuk** [PNU-SE-AIteam/cifar-10-stable-diffusion-detection#80]
Consider the following code:
```
for i in range(hp.Int('num_conv_layers', min_value=2, max_value=6, step=1)):   
        x = layers.Conv2D(filters=#...irrelevant...)(x)
        #...irrelevant...
        if hp.Boolean(f'CONV_{i}_pooling'):
            x = layers.MaxPool2D()(x)
        if hp.Boolean(f'CONV_{i}_norm'):
            x = layers.BatchNormalization()(x)
```
For example, ```num_conv_layers == 4```. Then the amount of MaxPool2D layers in this block is a binomial distribution:
- 0.0625 chance to have **0** MaxPool2D
- 0.25 chance to have **1** MaxPool2D
- 0.375 chance to have **2** MaxPool2D
- 0.25 chance to have **3** MaxPool2D
- 0.0625 chance to have **4** MaxPool2D
![image](https://github.com/PNU-SE-AIteam/cifar-10-stable-diffusion-detection/assets/126328450/c1c4a7b8-7e5b-469a-b234-84d4a920ed2c)
This is an issue because the amount of trials in Hyperband is incomparably small in contrast to the number of other hyperparameter combinations. 
If the optimal MaxPool2D amount were to be **0**, but there were only 3 "good" trials left from all the other hyperparameters, there is a considerable chance that we never even *choose* "zero MaxPool2D" because of how unlikely it is.
This hypothesis does appear to be the case (at least partly) in our actual model with both MaxPool2D (seems to be more inclined to lower numbers) and BatchNormalization (seems to be more inclined to higher numbers). 

Also, even if we know the optimal amount of each layer, we cannot shift our hyperparameter space in that direction since we're just using boolean hyperparameters.

**Tasks:**

Turn the *binomial* distribution of parameters in a for loop into a *uniform* distribution.

Make these parameters tuneable.
